### PR TITLE
Han/refactoring and aspnetcore updates

### PR DESCRIPTION
**Pull request by hanwavefront**
- Added ApplicationTags and HeartbeaterService
- Renamed everything to Wavefront.SDK.CSharp
- Updated readme
- Depends on WavefrontHistogram (for ASP.NET Core SDK), which hasn't yet been merged to master

### CODE REVIEW

**Comment by vikramraman** [comment]
There are some README merge conflicts. Can you please resolve them? lgtm otherwise.

**Comment by hanwavefront** [comment]
Fixed merge conflict. I believe we need approval of WavefrontHistogram before merging.

**Review by sushantdewan123 — APPROVED** [review]
LGTM

**Pull request merged**
